parser/src/main/java/me/vertretungsplan/parser/LegionBoardParser.java

```diff
@@ -119,7 +120,12 @@ private JSONArray getTeachers() throws IOException, JSONException {
 	}
 
 	private JSONArray getJSONArray(String url) throws IOException, JSONException {
-		return new JSONArray(httpGet(url, "UTF-8"));
+		try {
+			return new JSONArray(httpGet(url, "UTF-8"));
+		} catch (HttpResponseException e) {
+			e.printStackTrace();
```

Comment:
Well, even though I (and many other people) do it often, just catching exceptions and printing their stack trace is a bad habit, especially if you actually expect these exceptions to occur sometimes (such as when the server is down). If you need no special handling of HTTP error codes, I would just remove the try-catch block and let it throw the `HttpResponseException` (which is a subclass of `IOException`).